### DISCUSSION TITLE
Strange issue with uploading screenshots on Android

### DISCUSSION THREAD

**Discussion by jbryant314** [GetStream/stream-chat-flutter#403] [Q&A]
Hello, I've encountered a strange issue and I'm not sure where to even begin troubleshooting.  When uploading a **screenshot** the first time on Android, the upload completes but when trying to load the URI returned I get an HTTP `422` and the image does not load in chat.  If I upload the exact same screenshot again, it works as expected.  This doesn't happen with photos taken from the camera, whether I choose them from the image library or using the camera directly (from attachment options).

URIs returned from uploading the exact same screenshot:

First (fails): https://stream-chat-us-east-c4.imgix.net/105610/images/f88c3b0c-55e9-426e-a705-79eb017de213.Screenshot_20210422-185230.png?ro=0&s=eb03147e3dc7e35e9e0dda58bd4a4553

Second (works): https://stream-chat-us-east-c4.imgix.net/105610/images/831f48bd-86c4-4801-b8d4-8e4e06ed12c0.Screenshot_20210422-185230.png?ro=0&s=44a9ddbf744add0c348ee1bed18dbb33

If you remove the query string from the first URI, it does load.  I've debugged this watching the upload calls in `message_input.dart` and I wasn't able to see any difference in the first and second API calls to upload the image.

Please help, I'm totally lost on this.  App ID: 105610.  Thanks!

Screen recording: https://photos.app.goo.gl/8PFr6ovKocooUYKU6

Note I'm using `stream_chat_flutter: ^0.2.21+1` (waiting for null safety to upgrade).  But this feels like it could be a backend problem?

Phone (happens on all my Android test devices)
Pixel 4a
Version: 11
Build Number: RQ2A.210405.005

**Comment by d3xvn** [comment]
Hey @futebolbryant,

Testing on a physical Android device on the latest stream_chat_flutter version, this bug does not seem to appear for me.
Any reason you can't upgrade to the latest version?
Things should work even before null safety comes along. 

Thanks.
